index.bs

```diff
@@ -365,76 +380,123 @@ spec: webidl
   </section>
 </section>
 <section>
-  <h2 dfn-type="dfn" export>
-    Permission Store
-  </h2>
+  <h2 id="permission-stores">Permission stores</h2>
   <p>
-    <a lt='user agent'>User agents</a> MAY use a form of storage to
-    keep track of web site permissions. When they do, they MUST have a
-    <dfn export>permission storage identifier</dfn> which is linked to a
-    {{PermissionStorage}} instance or one of its subtypes.
+    Each <a>realm</a> has a <dfn export>permission store</dfn> consisting of,
+    for each {{PermissionName}}, either a <dfn>denied flag</dfn> or a set
+    of <a>permission entries</a>. <dfn>Permission entries</dfn> for a
+    {{PermissionName}} are instances of that {{PermissionName}}'s <a>permission
+    entry type</a>, which will be {{PermissionStorageEntry}} or one of its
+    subtypes. The <a>permission entry type</a> specifies the <a>equivalent
+    permission entry</a> relation used to de-duplicate the <a>permission
+    entries</a>.
   </p>
+
   <p>
-    To <dfn>get a permission storage identifier</dfn> for a
-    {{PermissionName}} <var>name</var> and an <a>environment settings
-    object</a> <var>settings</var>, the UA MUST return a tuple consisting
-    of:
+    The UA may initialize <a>realm</a>'s <a>permission store</a> almost
+    arbitrarily, subject to the constraints in <a
+    href="#permission-store-constraints"></a>. The UA may also <a>queue a
+    task</a> to update a <a>realm's</a> <a>permission store</a> almost
```

Comment:
Making the UA queue tasks to update the realm's permission store gives a guarantee that it doesn't change within a single turn/microtask checkpoint. However, that's not so valuable since functions like `query()` and `getCurrentLocation()` operate in parallel and so could be affected by a task posted while they're executing.

@raymeskhoury has suggested that we fall back to having *read* and *write* return completely UA-determined results. I think that's plausible. It would make it harder to write things like "the permission store only changes if the UA receives new information about the user's intent" here, but I think (e.g. in the media capture spec) "the UA must not return cameras entries from 'read the permission store' unless the user has explicitly given permission for the origin to access them" still works.

What do folks think?